### ISSUE TITLE
xbps-remove: failed to remove directory "No space left on device"

### ISSUE THREAD

**Issue by wine** [voidlinux/xbps#255]
Today I wanted to remove the old 4.11 kernel headers from my system:

`xbps-remove linux4.11-headers`

But xbps-remove showed me this:

```
Removing `linux4.11-headers-4.11.12_1' ...
ERROR: linux4.11-headers-4.11.12_1: failed to remove directory `/usr/src/kernel-headers-4.11.12_1/include/config/panic/on': No space left on device
ERROR: linux4.11-headers-4.11.12_1: failed to remove directory `/usr/src/kernel-headers-4.11.12_1/include/config/dmi/scan/machine/non/efi': No space left on device
ERROR: linux4.11-headers-4.11.12_1: failed to remove directory `/usr/src/kernel-headers-4.11.12_1/include/config/dns': No space left on device
ERROR: linux4.11-headers-4.11.12_1: failed to remove directory `/usr/src/kernel-headers-4.11.12_1/include/config/rmi4/2d': No space left on device
ERROR: linux4.11-headers-4.11.12_1: failed to remove directory `/usr/src/kernel-headers-4.11.12_1/include/config/rmi4/f03': No space left on device
...
```

...and so on. I also noticed an unusual hard drive activity which lasted for about five minutes, so I stopped xbps with CTRL+C and tried to remove the package again. The second time, xbps removed it instantly but with many "No such file or directory" errors, surely caused by the previous incomplete removal.

The issue is always reproducible on my system by installing and removing the linux headers package. I remember seeing this also while removing the 4.10 kernel some time ago. I think it has to do with my Btrfs root filesystem, but I never had any problem installing/removing/using other packages. Moreover, if I do this:

`rm -rf /usr/src/kernel-headers-4.11.12_1`

there are no errors. So I'm opening this issue here, because it seems to happen only with xbps.

**Comment by chneukirchen** [comment]
Which filesystem is that?

**Comment by wine** [comment]
My / is Btrfs with lzo compression enabled. I usually setup my systems this way and I never seen errors like those before in Void nor other distros, except while removing the linux headers package with `xbps-remove`. Maybe the relatively high number of files cointained in the package triggers some strange bug somewhere inside or outside xbps or Btrfs...

As I said above, using recursive `rm` works without errors.

**Comment by chneukirchen** [comment]
All I can say is don't run btrfs on full file systems...

**Comment by wine** [comment]
I did further tests and the problem is specific to my system and definitely NOT caused by xbps.

Sorry for the waste of time and thank you for building the best Linux distribution I've used in years.

**Comment by the-maldridge** [comment]
For the benefit of future readers, what was the problem and how did you solve it?

**Comment by wine** [comment]
I have not "solved" it really, otherwise I would have written the solution here... I just tried to reproduce the issue in another Void installation with a very similar setup (btrfs root with lzo compression, etc) and the problem there simply does not exist. Probably the errors are related to my btrfs root filesystem and not to xbps.

Sorry again.

**Comment by the-maldridge** [comment]
Ah, I will keep an eye out for this on my btrfs systems.  I know that xbps can get btrfs into a situation where you need to manually invoke a tree re-balance, but that shouldn't be very often.

**Comment by wine** [comment]
I tried the btrfs balance commands, but nothing changed with xbps.

Thinking that btrfs on the affected system was corrupted, I've tried to recreate it using the same settings. Then I've done a filesystem-level copy of the Void installation from a backup. The errors are still there and I don't know what else I can try to detect the cause. Root is full only for 30% now...

Reopen the issue if you think it's worth it...
